src/WebJobs.Extensions.MobileApps/Config/MobileAppsConfiguration.cs

```diff
@@ -64,84 +67,40 @@ public void Initialize(ExtensionConfigContext context)
 
             string uriString = _nameResolver.Resolve(AzureWebJobsMobileAppUriName);
             Uri.TryCreate(uriString, UriKind.Absolute, out _defaultMobileAppUri);
+            
+            var rule = context.AddBindingRule<MobileTableAttribute>();
+            rule.AddValidator(ValidateMobileAppUri);
 
-            BindingFactory factory = new BindingFactory(_nameResolver, converterManager);
+            rule.BindToCollector<OpenType>(typeof(MobileTableCollectorBuilder<>), this);
+            rule.BindToInput<IMobileServiceClient>(new MobileTableClientBuilder(this));
 
-            IBindingProvider outputProvider = factory.BindToCollector<MobileTableAttribute, OpenType>(typeof(MobileTableCollectorBuilder<>), this);
+            rule.WhenIsNotNull(nameof(MobileTableAttribute.TableName)).
```

Comment:
Mobile has some pretty aggressive rules here that combine Attribute and parameter type (which I think . This is using the fluent declarative filtering - it detangles it and should be preserving the same semantics as the original. 

Note that  we no longer need per-rule predicates. We can use the OpenType system and just say IMobileServiceTableQuery<MobileType> and IMobileServiceTable<MobileType>

Reply:
I believe there are enough tests for all of this, so if they pass, we should feel confident that the rules match.